tools/alive_parser.cpp

```diff
@@ -849,17 +873,29 @@ static unique_ptr<Instr> parse_call(string_view name) {
   return call;
 }
 
+static unique_ptr<Instr> parse_malloc(string_view name) {
+  // %p = malloc ty %sz
+  auto &ty = parse_type();
+  auto &op = parse_operand(ty);
+  // Malloc returns a pointer at address space 0
+  Type &pointer_type = get_pointer_type(0);
+  return make_unique<Malloc>(pointer_type, string(name), op);
+}
+
 static unique_ptr<Instr> parse_copyop(string_view name, token t) {
   tokenizer.unget(t);
   auto &ty = parse_type();
   auto &op = parse_operand(ty);
   return make_unique<UnaryOp>(ty, string(name), op, UnaryOp::Copy);
 }
 
-static unique_ptr<Instr> parse_instr(string_view name) {
+static unique_ptr<Instr> parse_instr(optional<string_view> nameopt) {
```

Comment:
I guess here you can just pass an empty string_view and avoid all the changes below.
Also, the code crashes if not given a register when expected. This needs checking.

Reply:
Ok, makes sense.

Reply:
Done